proparse/src/main/java/org/prorefactor/proparse/antlr4/Lexer.java

```diff
@@ -990,14 +1019,16 @@ void macroDefine(int defType) throws IOException {
     defText = defText.trim();
     // Do listing before lowercasing the name
     prepro.getLstListener().define(textStartLine, textStartCol, macroName.toLowerCase(Locale.ENGLISH), defText,
-        defType == ProParserTokenTypes.AMPGLOBALDEFINE ? MacroDef.GLOBAL : MacroDef.SCOPED);
-    if (defType == ProParserTokenTypes.AMPGLOBALDEFINE)
+            defType == PreprocessorParser.AMPGLOBALDEFINE ? MacroDef.GLOBAL : MacroDef.SCOPED);
+    if (defType == PreprocessorParser.AMPGLOBALDEFINE)
       prepro.defGlobal(macroName.toLowerCase(), defText);
     else
       prepro.defScoped(macroName.toLowerCase(), defText);
   }
 
   void macroUndefine() throws IOException {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.io.IOException', as it cannot be thrown from method's body. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)